src/data_structures/README.md

```diff
@@ -80,4 +80,34 @@ __Source to read:__
 
 [doubly-linked-list]: https://upload.wikimedia.org/wikipedia/commons/thumb/5/5e/Doubly-linked-list.svg/610px-Doubly-linked-list.svg.png
 
-[stack]: https://upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Lifo_stack.png/700px-Lifo_stack.png
\ No newline at end of file
+[stack]: https://upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Lifo_stack.png/700px-Lifo_stack.png
+
+### [Queue](./queue.rs)
+
+- A queue is a `linear` data structure that is open at both ends with operations performed in *First in first out* (FIFO) or *Last in Last out* (LILO)
+- A queue is considered a list where all additions are made on one end and all deletions on the opposite end to adhere to the FIFO principle
+
+#### Queue principles
+- The `front` is considered the location where the first entry will be removed from the queue
+- The `rear` is considered the location of the latest entry and will be the last entry removed from the queue
+
+![image](https://media.geeksforgeeks.org/wp-content/uploads/20220805131014/fifo.png)
+
+#### Queue Methods
+- `Enqueue()`:  items are added to the rear, returns the value of the node or item added
+- `Dequeue()`:  items are removed from the front, returns the value of the node or item removed
+- `Peek()`: returns the value of the front, throws an exception if empty
+- `isEmpty()`: check if there is a node or item in queue returns a boolean
+
+#### Queue Properties
+- If the underlying data structure of a queue is a linked list:
+  - enqueue: O(n-i)
+  - peek: O(1)
+  - dequeue: O(n-i)*
```

Comment:
```suggestion
  - enqueue: O(1)
  - peek: O(1)
  - dequeue: O(1)*
```

The cost of `enqueue` and `dequeue` is constant since in both a linked list and an array, you can do `O(1)` operations.

- Linked List: You maintain a pointer to the `front` and the `rear`. `dequeue` is taking the first element, removing its pointer to the previous element, the pointer from the previous to the first, and then make the previous the front. `enqueue` is analog.
- Array: Most languages have built-in `push` and `pop` operations in an array-like data structure.

Also, you must always define your variables first so that the reader talks in the same term as you do. That is, you have to define `n` in this case, which I would do as "where `n` is the number of elements present in the Queue".